### PR TITLE
BP-3316 - Credit cards brands seperate in the checkout

### PR DESCRIPTION
**Pull request by LucianTuriacArnia**

### CODE REVIEW

File: view/frontend/web/js/view/payment/method-renderer/creditcard.js [hunk 1]
```diff
@@ -155,13 +181,18 @@ define(
                     return true;
                 },
 
+                isCredicardGroupMode: function () {
+                    return window.checkoutConfig.payment.buckaroo.creditcard.groupCreditcards !== undefined && window.checkoutConfig.payment.buckaroo.creditcard.groupCreditcards == 1 ? true : false;
```
Comment (harli91) [inline view/frontend/web/js/view/payment/method-renderer/creditcard.js:185]: why not do `return window.checkoutConfig.payment.buckaroo.creditcard.groupCreditcards === 1`

File: view/frontend/web/js/view/payment/method-renderer/creditcard.js [hunk 2]
```diff
@@ -155,13 +181,18 @@ define(
...
+                },
+
                 getData: function () {
                     var selectedCardCode = null;
                     if (this.selectedCard()) {
-                        selectedCardCode = this.selectedCard().code;
+                        selectedCardCode = typeof this.selectedCard() === 'object' ?
+                            this.selectedCard().code :
+                            this.selectedCard();
```
Comment (harli91) [inline view/frontend/web/js/view/payment/method-renderer/creditcard.js:193]: i think its better here if `selectedCard` return a single type

File: view/frontend/web/template/payment/buckaroo_magento2_creditcard.html [hunk 3]
```diff
@@ -97,3 +98,71 @@
...
+                        <!-- ko text: name --><!-- /ko -->
+                    </label>
+                </div>
+                <div class="bk-img-wrap">
+                    <img data-bind="attr: { src: img }"/>
+                </div>
+            </div>
+        </div>
+
+        <div class="payment-method-content">
```
[the remark this answers is not in the document]
Comment (LucianTuriacArnia) [reply view/frontend/web/template/payment/buckaroo_magento2_creditcard.html:125]: I think it's necessary to allow customers to see error messages and to have the option to modify their billing address accordingly. This is different from the case with gift cards, where the order isn't finalized upon applying the gift card.

File: view/frontend/web/js/view/payment/method-renderer/creditcard.js [hunk 4]
```diff
@@ -147,6 +152,27 @@ define(
...
+                    this.item.method = 'buckaroo_magento2_creditcard';
+                    this.paymentMethod = this.item.method;
+                    window.checkoutConfig.buckarooFee.title('Fee');
+                    selectPaymentMethodAction({
+                        "method": this.item.method,
+                        "po_number": null,
+                        "additional_data": {
+                            "card_type" : code
+                        }
+                    });
```
Comment (harli91) [inline view/frontend/web/js/view/payment/method-renderer/creditcard.js:171]: why not call `this.selectPaymentMethod()` ?

File: view/frontend/web/js/view/payment/method-renderer/creditcard.js [hunk 5]
```diff
@@ -107,6 +108,10 @@ define(
                     return this;
                 },
 
+                getCreditcardType: ko.observable(function () {
+                    return this.creditcardIssuer;
+                }),
```
Comment (harli91) [inline view/frontend/web/js/view/payment/method-renderer/creditcard.js:113]: what's the use of this observable?

**Review by harli91 — CHANGES_REQUESTED** [review]

File: view/frontend/web/js/view/payment/method-renderer/creditcard.js (hunk 5 above)
Comment (LucianTuriacArnia) [reply view/frontend/web/js/view/payment/method-renderer/creditcard.js:113]: it is used in <div data-bind="attr: { 'class': 'buckaroo_magento2_all_creditcard payment-method buckaroo_magento2_' + code }, css: {'_active': ($parent.getCode() == $parent.isChecked() && **$parent.getCreditcardType()** == code)}">

This adds a CSS class _active if the credit card type returned by getCreditcardType() matches the code variable

Comment (harli91) [reply view/frontend/web/js/view/payment/method-renderer/creditcard.js:113]: why not use directly `selectedCard()` method?

**Review by harli91 — DISMISSED** [review]

**Pull request merged**